fix(components): [tooltip] fix active element

danny-hebert opened the pull request:
When the tooltip is used inside Vue Custom elements, the document.activeElement is not returning the good element.
This fix makes sure to get the real active element including shadowDom if present.
The select with the option Multiple would always close the dropdown each time you select a value.

This allows users to use Element plus v-click-outside date-picker, time-picker inside Vue Custom elements.
Without this change, it's impossible to select a date or a time in the date-picker and time-picker inside a Custom element.

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

danny-hebert committed:
fix(components): [tooltip] fix active element

consultation-applio commented:
Playground with the new working version :
https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCB2YWx1ZTEgPSByZWYoW10pXG5cbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjEnLFxuICAgIGxhYmVsOiAnT3B0aW9uMScsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjInLFxuICAgIGxhYmVsOiAnT3B0aW9uMicsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjMnLFxuICAgIGxhYmVsOiAnT3B0aW9uMycsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjQnLFxuICAgIGxhYmVsOiAnT3B0aW9uNCcsXG4gIH0sXG4gIHtcbiAgICB2YWx1ZTogJ09wdGlvbjUnLFxuICAgIGxhYmVsOiAnT3B0aW9uNScsXG4gIH0sXG5dXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8aDM+XG4gICAgU3RhbmRhbG9uZSB0b29sdGlwXG4gIDwvaDM+XG4gIDxlbC10b29sdGlwXG4gICAgICBjbGFzcz1cImJveC1pdGVtXCJcbiAgICAgIGVmZmVjdD1cImRhcmtcIlxuICAgICAgY29udGVudD1cIlRvcCBMZWZ0IHByb21wdHMgaW5mb1wiXG4gICAgICBwbGFjZW1lbnQ9XCJ0b3Atc3RhcnRcIlxuICAgICAgOnRlbGVwb3J0ZWQ9XCJmYWxzZVwiXG4gICA+PGVsLWJ1dHRvbj50b3Atc3RhcnQ8L2VsLWJ1dHRvbj5cbiAgIDwvZWwtdG9vbHRpcD5cbiAgPGgzPlxuICAgIFRvb2x0aXAgaW4gc2VsZWN0IHVzaW5nIHRoZSBwcm9wcGVyIGFjdGl2ZUVsZW1lbnRcbiAgPC9oMz5cbiAgPGVsLXNlbGVjdFxuICAgIHYtbW9kZWw9XCJ2YWx1ZTFcIlxuICAgIG11bHRpcGxlXG4gICAgcGxhY2Vob2xkZXI9XCJTZWxlY3RcIlxuICAgIHN0eWxlPVwid2lkdGg6IDI0MHB4XCJcbiAgICA6dGVsZXBvcnRlZD1cImZhbHNlXCJcbiAgPlxuICAgIDxlbC1vcHRpb25cbiAgICAgIHYtZm9yPVwiaXRlbSBpbiBvcHRpb25zXCJcbiAgICAgIDprZXk9XCJpdGVtLnZhbHVlXCJcbiAgICAgIDpsYWJlbD1cIml0ZW0ubGFiZWxcIlxuICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgLz5cbiAgPC9lbC1zZWxlY3Q+XG5cdFxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBFcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG48L3RlbXBsYXRlPlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTEyMDU5LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTIwNTktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctMTIwNTktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTIwNTktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=

makedopamine committed:
fix(components): [table] width of columns become 0 unexpectedly (#12920)

zhengsixsix committed:
fix(components):[time-picker] add parseFloat to resolve TimePicker offset (#12837)

* fix(components): add parseFloat to resolve TimePicker offset

Height is calculated using offsetHeight, offsetHeight will be rounded off

closed #12835

* fix(components): import getStyle to optimize the code

closed #12835

* Delete useless statements

* add judgment

* Update basic-time-spinner.vue

wzc520pyfm committed:
docs(components): [time-picker] use new display tag (#12691)

* docs(components): [time-picker] use new display tag

* docs(components): [time-picker] update default of attributes

wzc520pyfm committed:
fix(components): [input] fix formatter and parse problem (#12622)

* fix(components): [input] fix formatter and parse problem

closed #12613

* fix(components): [input] fix input value problem

* fix(components): [input] update return value for input event emit

* test(components): [input] add test for formatter and parser

CeceWall committed:
perf(components): [select-v2] performance enhancement for initStates (#12390)

closed #12274

Co-authored-by: weijian <wudizhanche1000@gmail.com>

wzc520pyfm committed:
docs(components): [switch] use new display tag (#12678)

* docs(components): [switch] use new display tag

* style(components): [switch] update the batch deprecate

* docs(components): [switch] delete deprecated attributes

zwgwf committed:
style(theme-chalk): remove unreferenced modules (#12974)

tolking committed:
feat(theme-chalk): [form] add default width to Input in inline form (#12897)

* feat(theme-chalk): [form] add default width to Input in inline form

* docs: add clearable attributes to Input

* docs: add a description of the Input width

wuyadan committed:
docs(components): [progress] mark the versions of striped & striped-flow (#13116)

Co-authored-by: ut002528 <wuyadan@uniontech.com>

chenxch committed:
docs(components): [dialog] add deprecated describe (#13143)

mdoi2 committed:
fix(components): [table] fix clearFilter type (#13149)

N-M committed:
fix(locale) : improve sr translation (#13125)

Improve sr (Serbian) translation

yj-liuzepeng committed:
style(theme-chalk): change input-number hover style (#13120)

zhengsixsix committed:
style(theme-chalk): change border-left:0 to border-left-color:transparent (#13107)

fix(theme-chalk): change border-left:0 to border-left-color:transparent

closed #13089

ModyQyW committed:
fix(components): [form] improve types (#12549)

* fix(components): [form] improve types

* fix(components): [form] improve FormRules type, support object and array

* fix(components): [form] improve FormRules type with comment

* fix(components): [form] allow number as key in FormRules type

* chore(components): [form] fix FormRules type example

* chore: remove extra import

* chore: update form/custom-validation and form/validation examples

* chore(components): [form] update example

wuyadan committed:
docs(components): [select-v2] correct the description of placeholder (#13200)

Co-authored-by: ut002528 <wuyadan@uniontech.com>

ModyQyW committed:
fix(components): [calendar] improve types (#13165)

chenxch committed:
fix(components): [select] filter shake (#12106)

* fix(components): [select] filter shake

* fix(components): [select] filter shake

* fix(components): [select] filter shake

* fix(components): [select] filter shake

* fix(components): [select] remove tagInMultiLine

yj-liuzepeng committed:
fix(components): [tag] element vertically centered (#13235)

fix(components): [tag] fix bug #13229

sleepyShen1989 committed:
test(components): [cascader] change way to get defineExpose data (#13227)

cascaderRef.value is undefined

closed #13226

btea committed:
feat(components): [upload] item add title attribute (#13231)

yuyinws committed:
feat(components): [Table] add show-overflow-tooltip props (#13169)

* feat(components): [Table] add show-overflow-tooltip props

* docs(components): [Table] add usage for show-overflow-tooltip

* fix(components): [TableColumn] props priority

* test(components): add unit test for table

0song committed:
chore: update doc site (#13291)

btea committed:
fix(components): [table] adjust the showOverflowTooltip value logic (#13282)

* fix(components): [table] adjust the showOverflowTooltip value logic

* fix: set showOverflowTooltip default value to undefined

element-bot committed:
chore: Update changelog 2.3.7 (#13321)

chore: update changelog 2.3.7

btea committed:
test: [table] add show-overflow-tooltip to table and table-column (#13323)

yj-liuzepeng committed:
docs(components): [image] update image slot viewer description (#13325)

wonderl17 committed:
fix: correct sass map has-key parameter (#13132)

fix: correct map.has-key using

vaebe committed:
docs(components): [dialog] add description of props modal-class (#13122)

HeftyKoo committed:
fix(components): [date-picker] One input affect the other input (#12181)

* fix(components): [date-picker] One input affects the other input

* fix: check if minDate or maxDate is empty

* chore: remove debugger

Commit:
fix(components): [virtual-list] Resetting the vertical scrollbar state (#13319)

wzc520pyfm committed:
refactor(components): [select] override the class using computed (#13302)

* refactor(components): [select] override the class using computed

* refactor(components): [select] shorthand class object

init-qy committed:
fix(components): [tree] fix node can drop inner (#9770)

btea committed:
fix(style): [progress] linecap value of track is consistent with path (#13395)

fix(style): [progress] linecap value of track is consistent with circle

0song committed:
chore: update doc (#13434)

tolking committed:
fix(theme-chalk): [form] remove default width of Input in inline form (#13375)

0song committed:
chore: update doc (#13435)

0song committed:
chore: update doc (#13440)

chenxch committed:
fix(components): [checkbox] default after border color is transparent (#13402)

btea committed:
fix(components): [table] multiple line ellipsis tooltip takes effect (#12805)

wuzhidexiaoming committed:
fix(style): [select] tooltip positioning is incorrect (#13424)

fix(style): [select] tooltip positioning is incorrect (#13251 #13422)

YunYouJun committed:
fix: tree content checkbox height larger than item height, close #13396 (#13446)

754664196 committed:
docs: [progress] remove useless style (#13465)

docs(docs):circular-progress-bar文档出现了多余的css信息

Co-authored-by: HeZhi <zhi.he@caijai.com>

btea committed:
docs: tooltip set max-width (#13462)

btea committed:
chore: format code (#13467)

btea committed:
docs(components): [message-box] (#13470)

yj-liuzepeng committed:
docs(components): [tree-v2] add some disabled nodes (#13484)

* docs(components): [tree-v2] add some disabled nodes

* docs(components): [tree-v2] add some disabled nodes

zhuying547 committed:
fix: types in exports (#13443)

sxzz committed:
fix: add more exports types (#13498)

kooriookami committed:
fix(components): [color-picker] fix incorrect cursor position (#13516)

fix(components): [color-picker] fix incorrect cursor position. Closes #12581 #12901

sxzz committed:
chore: update renovate config (#13536)

kaine0923 committed:
style(theme-chalk): [table] fix th font weight (#13556)

style(theme-chalk): [table] change th font weight to 600

style(theme-chalk): [table] remove tag style files

Co-authored-by: kaine <kainewan@163.com>

tolking committed:
fix(components): [input] lose focus when click suffix icon (#13264)

* feat(hooks): add useFocusController

* fix(components): [input] lose focus when click suffix icon

closed #13153, #13159

* refactor(hooks): use-focus-controller

* test: fix test

* test: update

* fix(theme-chalk): [input] add cursor style to the wrapper

element-bot committed:
chore: Update changelog 2.3.8 (#13592)

chore: update changelog 2.3.8

btea committed:
refactor(components): [input] extract InputInstance (#13620)

shigma committed:
fix(components): [input] fix textarea disabled box-shadow (#13632)

Update input.scss

btea committed:
docs: [input] list the acceptable values of type (#13644)

Commit:
docs(components): [table-v2] update the type description for Events (#13362)

kaine0923 committed:
fix(components): [collapse-transition] fixed transition canceled bug (#13358)

closed #13357

wzc520pyfm committed:
docs(components): [dialog] use new display tag (#13499)

* docs(components): [dialog] use new display tag

* chore(components): [dialog] remove the incorrect deprecated comment

matto49 committed:
fix(components): [tree] fix expand-icon.is-leaf prevent click event (#13616)

btea committed:
chore(docs): format message-box (#13676)

tolking committed:
docs(build): add required attribute to the tag plugin (#13691)

qiYuei committed:
fix(components): [focus-trap] Fixed the function that removed the listener (#13158)

fix(components): [focus-trap]

correct-error-listening

closed #13128

Co-authored-by: fengmingjian <fmj@hanzhisoft.com>

wonderl17 committed:
fix(components): [table] add filteredValue watcher (#13121)

init-qy committed:
fix(components): [autocomplete] improve a11y (#12647)

* fix(components): [autocomplete] improve a11y

* fix(components): add a11y tests

* fix: lint

hhucqian committed:
refactor(components): [upload] fix the type of file (#13637)

change the type of UploadRequestOptions.file from File to UploadRawFile

btea committed:
docs(components): [drawer] (#13677)

wuzhidexiaoming committed:
fix(components): [select] fix for confusing click behaviour (#13664)

fix(components): [select] fix for confusing click behaviour (#13642)

tolking committed:
refactor(components): [menu] adjust default value of fallbackPlacements (#13696)

closed #13682

tolking committed:
fix(theme-chalk): [menu] the popup menu has 2px border at the bottom (#13695)

yj-liuzepeng committed:
fix(components): [row] align="top" is not working properly (#13550)

* fix(components): [row] align='top' is not working properly

* fix(components): [row] align="top" is not working properly

yj-liuzepeng committed:
docs(components): [row] remove the default value of align (#13562)

makedopamine committed:
chore: prevent warnings while running test cases (#12189)

* chore: prevent warnings while running test cases

* fix: overlook a test case

MrWeilian committed:
fix(components): [cascader] reactive size change (#12294)

* fix(components): [cascader] reactive size change

* fix(components): [cascader] update style after calc

* fix(components): [cascader] add test

* fix(components): [cascader] mock offsetHeight

* fix(components): [cascader] size type

* fix(components): [cascader] spy on getter for height

* fix(components): [cascader] mock css var

* fix(components): [cascader] add comment

mhpsy committed:
fix(components): [collapse-transition] remove redundant code (#13715)

fix(components): [collapse-transition]

tolking committed:
fix(hooks): [use-focus-controller] remove unnecessary role attributes (#13749)

fix(hooks): [a11y] remove unnecessary role attributes

evanryuu committed:
chore(components): [input] fix inputStyle comment typo (#13768)

makedopamine committed:
chore(components): [date-picker] export instasnce type (#13618)

* chore(components): [date-picker] export instasnce type

* move DatePickerInstance to instance.ts

btea committed:
fix(components): [date-picker] disable button if now is disabled (#13655)

* fix(components): [date-picker] disbale button if now is disabled

* fix: test

* chore: add test

hunter-fl committed:
test(components): [button] add test cases for using props of form and form-item (#13726)

test(components): add test cases for using props of form and form-item

add case for using size and disabled from form

add case for using size from form-item

btea committed:
feat(components): [switch] add custom action icon prop (#13746)

lxy030988 committed:
fix(components): [el-table] rangeWidth dealing with loss of precision (#12938)

* fix(components): [el-table] rangeWidth use Math.ceil

Closes #12687

* fix(components): [el-table] rangeWidth calculate

Closes 12938

* fix(components): [el-table] offsetWidth calculate

Closes #12938

Jungzl committed:
docs: align ::marker and anchor on the same line in firefox (#13752)

btea committed:
docs(components): [switch] add version tag & fix typo (#13771)

* docs(components): [switch] add version tag & fix typo

* chore: add tag

acyza committed:
fix(components): [cascader] checkbox sync error (#11462)

* fix(components): [cascader] sync error

* fix(components): [cascader] checkbox sync error

yj-liuzepeng committed:
fix(components): [avatar] set img tag's width attribute (#13844)

kooriookami committed:
Doc/figma (#13804)

* docs: update resource page

* docs: update

* docs: update

* docs: update

tolking committed:
fix(components): [select] abnormal focus when click tags (#13699)

* fix(components): [select] abnormal focus when click tags

closed #13665

* fix(components): [select] clearable

* chore(components): [select] remove console

* fix: the setTimeout function may bring some side effects

* fix: remove role

* test(components): [select] add some test

btea committed:
docs: logo and parallax add drop-shadow (#13833)

init-qy committed:
fix(components): [select] clearIcon did not work with form statusIcon (#13533)

* fix(components): [select] clearIcon not effect with form statusIcon

* fix(components): [select] add computed value

* fix(components): [select] add comment message

* fix(components): [select] add missing import

* refactor: [select] place selectTagsStyle in useSelect

* fix: [select] remove unused export

* fix: compress code

* fix: fix error

element-bot committed:
chore: Update changelog 2.3.9 (#13870)

chore: update changelog 2.3.9

btea committed:
docs: fix example ts error (#13849)

evanryuu committed:
chore(components): [date-picker] correct typo in test file (#13876)

Jungzl committed:
docs(components): [pagination] correct text & icon priority (#13743)

* docs(components): [pagination] correct text & icon priority

* docs(components): [pagination] improve description

btea committed:
feat(ci): enable provenance (#13799)

yj-liuzepeng committed:
fix(components): [tree] the issue with retrieving childrenKey (#13798)

* fix(components): [tree] the issue with retrieving childrenKey

* Update tree-node.vue

* Update tree-node.vue

btea committed:
test(components): [tree] custom label and children test case (#13902)

btea committed:
feat(ci): enable provenance (#13901)

JunIce committed:
docs: fix official migration reference link (#13911)

fix(docs): fix official migration reference link

tiansQAQ committed:
refactor(components): [form-label-wrap] use getStyle method (#13913)

cc-hearts committed:
fix(components): [select] fix set value error(#13872) (#13879)

btea committed:
refactor(components): [table] reuse rAF (#13918)

MilesTails01 committed:
fix(docs): [table-v2.md] corrected en-US grammar, spelling, readability (#13900)

* fix(docs): [table-v2.md] corrected en-US grammar, spelling, readability

* fix(docs): [table-v2.md] resolved feedback

* fix(docs): [table-v2.md] resolved feedback, further improvements

hunter-fl committed:
test(components): add test case for delay of dialog (#13773)

MrWeilian committed:
fix(components): [autocomplete] tooltip offset (#12216)

* fix(components): [autocomplete] tooltip offset

* fix(components): [autocomplete] remove nextTick

uxuip committed:
feat(components): [card]  add body class (#11869)

* feat(components): [card]  add body class

* fix(components): [card] remove default value

* fix(components): [card] modify description

btea committed:
refactor: [date-picker] rename time-arrow-control (#13651)

* refactor: [date-picker] remove time-arrow-control

* docs: rename time-arrow-control

Jackie-llv committed:
docs(components): [form] modify props description (#13950)

* docs(components): [form] update prop's type

* fix(components): [form] modify props description

---------

Co-authored-by: lvjiaqi <lvjiaqi@itcast.cn>

LiZheGuang committed:
fix(components): fix the problem that chrome 72 table doesnotdisplaydata (#10640)

Fix the problem that the chrome 72 tab table does not display data. This is caused by the fact that
the extension operator becomes an empty object in a special scenario

BREAKING CHANGE :
The writing method of shallow copy is replaced

closed #8770,#9071,#7038

Co-authored-by: btea <2356281422@qq.com>

Karolis-Stoncius committed:
fix(components): [table] add table selection checkbox aria-labels (#13908)

When checkboxes have no label, it causes accessibility issues. This commit adds aria-labels to
table's selection checkboxes

youpinyao committed:
fix(components): [upload] fix empty array error (#13490)

* fix(components): [upload] fix empty array error

在 components/upload/src/ajax.ts 73 行中，如果data 中key对应的value 是空数组的情况下`if (Array.isArray(value)) formData.append(key, ...value)`报错

* fix: reuse isArray

---------

Co-authored-by: btea <2356281422@qq.com>

MrWeilian committed:
fix(components): [select] backspace delete disabled option (#11995)

* fix(components): [select] backspace delete disabled option

* fix(components): [select] findLastIndex

* fix(components): [select] simple polyfill findLastIndex in test file

* fix(components): [select] add test for backspace

* chore: lint

yj-liuzepeng committed:
fix(components): [tree-v2] modify the condition for when the value of currentKey is 0 (#13935)

* modify the condition for when the value of currentKey is 0

* add a test case

* add a test case

* add a test case

* Update packages/components/tree-v2/__tests__/tree.test.ts

Co-authored-by: btea <2356281422@qq.com>

* Update packages/components/tree-v2/__tests__/tree.test.ts

Co-authored-by: btea <2356281422@qq.com>

* Update packages/components/tree-v2/__tests__/tree.test.ts

Co-authored-by: btea <2356281422@qq.com>

---------

Co-authored-by: btea <2356281422@qq.com>

btea committed:
test: [button] add comment resolve ts error (#13965)

* test: [button] add comment resolve ts error

* chore: update comment

makedopamine committed:
fix(components): [table-v2] calculate the width of MainTable in the same way (#13584)

* fix(components): [table-v2] calculate the width of MainTable in the same way

* fix: duplicate properties

FrontEndDog committed:
fix(components): [radio checkbox] fix click event triggered twice (#13936)

fix(components): [radio checkbox] @click event triggered twice

kamesan012 committed:
fix(components): fix the attribute conflict (#13504)

* fix(components): fix the attribute conflict

When value is input by shortcut, it's value will not be changed by defaultTime anymore

closed #13483

* fix(components): change param to pass the test

* fix(components): delete unnecessary changes

* fix(components): use let to replace ref variable

* fix(components): add test cases

btea committed:
chore(components): [autocomplete] fix the comment description error (#13979)

0song committed:
chore: update readme (#13988)

btea committed:
chore: add node compatibility badge (#13986)

wzc520pyfm committed:
docs(components): [select] use new display tag (#13301)

* docs(components): [select] use new display tag

* docs(components): [select] description and comment begins in lower case

* docs(components): [select] update type of prop and event

yj-liuzepeng committed:
chore(docs): updating the implementation of the theme switch button (#14014)

* chore(docs): updating the implementation of the theme switch button

* chore(docs): updating the implementation of the theme switch button

* chore(docs): updating the implementation of the theme switch button

tolking committed:
fix(components): [dropdown] adjust onItemLeave trigger (#13212)

fix(components): trigger focus change when the trigger includes hover

closed #13131

btea committed:
chore: format code (#14037)

HeftyKoo committed:
refactor(components): [upload] remove useless useSlot (#14053)

refactor(components): [upload] remove useless useSlots

inottn committed:
fix(components): [select] fix the logic of updating selected data (#13374)

* fix(components): [select] fix the logic of updating selected data

* chore: improve judgment condition

kaine0923 committed:
docs(components): add attr version tag (#13641)

1. docs(components): [cascader] add version tag for togglePopperVisible expose

2. docs(components): [color-picker] add version tag for show & hide expose

3. docs(components): [date-picker] add version tag for handleOpen & handleClose method

4. docs(components): [dialog] add version tag for align-center atrr

5. docs(components): [dropdown] add version tag for teleported attr

6. docs(components): [form] add version tag for scroll-into-view-options attr

7. docs(components): [image] add version tag for zoom-rate attr

8. docs(components): [input-number] add version tag for readonly attr

9. docs(components): [menu] add version tag for popper-effect attr

10. docs(components): [message-box] add version tag for append-to attr & message function

11. docs(components): [rate] add version tag for clearable attr

12. docs(components): [select-v2] add max-collapse-tags attr

13. docs(components): [select] add version tag for collapse-tags-tooltip & placement

14. docs(components): [table] fix show-overflow-toolip attr version tag

15. docs(components): [time-picker] add version tag for handleOption & handleClose expose

16. docs(components): [tree-v2] add version tag for item-size attr

17. docs(components): [tree] add version tag for empty slot

inottn committed:
fix(components): [date-picker] fix the logic of checking the date range (#13377)

wzc520pyfm committed:
docs(components): [transfer] use new display tag (#13033)

* docs(components): [transfer] use new display tag

* docs(components): [transfer] update TransferKey to Array<string|number>

* docs(components): [transfer] directly specify type for TransferDataItem

chenweiyi committed:
feat(components): [cascader] add max-collpase-tags prop  (#13386)

* feat(components): add max-collapse-tags prop

show more than one tag in cascade when multiple enable and use collapse-tags true

* test(test-utils): add unit test for cascader prop 'max-collapse-tags'

* Update packages/components/cascader/src/cascader.vue

Co-authored-by: btea <2356281422@qq.com>

* Update cascader.vue

* refactor(components): [cascader] style lint

---------

Co-authored-by: chenweiyi01 <chenweiyi01@zuoyebang.com>
Co-authored-by: btea <2356281422@qq.com>

evanryuu committed:
fix(components): [date-picker] correct onCalendarChange param type (#13976)

Karolis-Stoncius committed:
feat(components): added aria-level attributes for dialog and drawer headers; Fixed input-number's aria-valuenow being null (#13822)

* feat(components): [dialog] added dialog header aria-level attribute

To pass accessibility tests elements with role="heading" require aria-level

* feat(components): [drawer] added drawer header aria-level attribute

To pass accessibility tests elements with role="heading" require aria-level

* fix(components): [input-number] prevent aria-valuenow from being null

To pass accessibility tests aria-valuenow cannot be null

* fix(components): [input-number] fix number input when 0 and update test

Fixes a bug where number input's aria-valuenow will be set to '' when input's value is 0. Also
updates a failing test

* refactor(components): [dialog] change default aria-level to 2

* docs(components): dialog and drawer header aria-level docs update

* docs(components): change headerAriaLevel type to string

* docs(components): kebab-case for docs

Karolis-Stoncius committed:
feat(components): [select] add aria-label to select's input (#13932)

* feat(components): [select] add aria-label to select's input

Select's input causes accessibility issues without aria-label, this commit adds it

* refactor(components): [select] change aria-label to label; update docs

* refactor(components): [select] change back to ariaLabel prop name

* docs(components): kebab-case for docs

btea committed:
docs: [cascader] add version tag (#14082)

element-bot committed:
chore: Update changelog 2.3.10 (#14096)

chore: update changelog 2.3.10

yj-liuzepeng committed:
docs: [card] add version tag (#14083)

jumtp committed:
chore(test): [message] wrong tag typo (#14099)

fix(test-utils): [message.test] wrong tag

btea committed:
Revert "refactor(components): [form-label-wrap] use getStyle method (… (#14106)

Revert "refactor(components): [form-label-wrap] use getStyle method (#13913)"

This reverts commit c6a6d9bc6e813ee05bb358a2e929f54d1144d9f8.

element-bot committed:
chore: Update changelog 2.3.11 (#14116)

chore: update changelog 2.3.11

sseinHu committed:
fix(components): [select] gurad value type error (#14074)

fix(components): [useSelect.ts] selected.slice is not a function

select In the case of radio, the packaged code will report select. slice is not a function, because the radio selected is an object, and the object does not have the api of slice

HeftyKoo committed:
refactor(components): [upload] reuse revkoeObjectURL function (#14126)

kinggq committed:
fix(components): [collapse] allow nested inputs in ElCollapseItem (#14061)

* fix(components): [collapse] allow nested inputs in ElCollapseItem to handle space key

* Update packages/components/collapse/src/collapse-item.vue

Co-authored-by: btea <2356281422@qq.com>

---------

Co-authored-by: btea <2356281422@qq.com>

wonderl17 committed:
docs: [select-v2] add item-height attribute into doc (#14033)

btea committed:
docs(components): [collapse] (#14066)

yj-liuzepeng committed:
fix(components): [rate] bind the textColor property (#14092)

* fix(components): [rate] bind the textColor property

* add test case

* add test case

* Update rate.test.tsx

* Update rate.md

btea committed:
chore(ci): modify nightly release `PKG_NAME` value (#14147)

btea committed:
refactor(components): [collapse] replace `keypress` with `keydown` (#14137)

FrankFang committed:
fix(components): [table] add tabindex to tbody to fix #13991 (#14020)

* fix(components): [table] add tabindex to tbody to fix #13991

* remove :focus style

* fix: remove the focus outline

* fix: replace focus with focus-visible

* chore: add comments to explain why the tabIndex attr is needed

ModyQyW committed:
fix(components): [form] improve types (#14062)

* fix(components): [form] improve types

* docs: update form

* fix: consider RegExp

* chore: update jsdoc

wzc520pyfm committed:
fix(components): [select] fix hover index error when creating option (#12635)

* fix(components): [select] fix error when creating option

closed #12634

* test(components): [select] add test for create and default first option

* fix(components): [select] compatible #11297

JBtje committed:
fix(i18n): improve Dutch(nl) translation (#14164)

TomatoDroid committed:
docs(components): message-box (#14179)

update message-box options

closed #14178

boomboy4 committed:
feat(components): [el-pagination] add teleported prop (#14072)

* fix(components): [pagination] 修复toplayer全屏模式下分页下拉框无法显示问题

分页组件新增teleported参数 默认为true  便于用户控制size下拉框是否插入body

closed #14070

* docs(docs): [pagination] Add a teleported attribute

closed #14070

* fix(components): [pagination] teleported props default value

delete set teleported default value

closed #14070

* fix(components): [pagination] change the default value of teleported

change the default value of teleported to true

* Update docs/en-US/component/pagination.md

Co-authored-by: qiang <qw13131wang@gmail.com>

---------

Co-authored-by: shaojia.cheng <shaojia.cheng@arvetech.net>
Co-authored-by: qiang <qw13131wang@gmail.com>

HADB committed:
feat(components): [image-viewer] Add rotate event when rotating images (#14138)

* feat(components): [image-viewer] Add rotate event when rotating images

* Update docs/en-US/component/image.md

Co-authored-by: btea <2356281422@qq.com>

---------

Co-authored-by: btea <2356281422@qq.com>

btea committed:
chore: format (#14200)

Karolis-Stoncius committed:
refactor(components): [date-picker] add aria-labels to date picker btns (#14149)

wonderl17 committed:
fix(components): [color-picker] first click should be trigger (#14209)

jianjunyuu committed:
feat(components): [upload] support async data (#14015)

* feat(components): [upload] support async data

* feat(components): [upload] support async data

* fix(components): [upload] unit test

* fix(components): [upload] unit test

* feat(components): [upload] update data docs

* fix(components): [upload] unit test

* chore: add version

---------

Co-authored-by: Hefty <yeyuqiudeng@gmail.com>

btea committed:
fix(docs): [date-picker] `calendar-change` event parameter type error (#14237)

btea committed:
docs: refactor switch theme (#14036)

HADB committed:
fix(docs): [upload] fix the version of supporting async data (#14242)

jw-foss committed:
refactor(components): [date-picker] basic-date-table (#10344)

* Extract logic out of basic-date-table file.

Karolis-Stoncius committed:
fix(components): [collapse] fix accessibility issues (#13832)

Using tab related aria attributes results in accessibility issues

Justineo committed:
fix(components): [date-picker] month range should emit `calendar-change` (#14262)

HeftyKoo committed:
fix(components): [input] missing autofocus types (#14246)

* fix(components): [input] missing autofocus types

closed #14243

* fix: autofocus default to false

element-bot committed:
chore: Update changelog 2.3.13 (#14286)

chore: update changelog 2.3.13

0song committed:
chore: update ci npm version (#14294)

element-bot committed:
chore: Update changelog 2.3.14 (#14296)

chore: update changelog 2.3.14

0song committed:
chore: update crowdin version to fix jdk error (#14300)

HeftyKoo committed:
fix(components): [menu] export MenuInstance (#14284)

* fix(components): [menu] export MenuInstance

closed #14251

* chore: export other component instance

* refactor: extract instace type to instance file

SaberA1ter committed:
fix(components): [carousel] Watch items causing state loss(#13010) (#13011)

* fix(components): [carousel] Watch items causing state loss(#13010)

* perf(components): [carousel] Using the third parameter of watch(#13010)

---------

Co-authored-by: ZZQChongYa <zhengziqiang@star-net.cn>

YunYouJun committed:
feat: add menu horizontal height for img center (#14307)

makedopamine committed:
docs(descriptions): fix the `customed style` example (#14310)

Karolis-Stoncius committed:
fix(components): [collapse] fix triggering form action when inside it (#14308)

When el-collapse with el-collapse-item is placed inside a form and the collapse trigger btn is
pressed, form's action as triggered. This fixes it

closed #14304

yj-liuzepeng committed:
docs: [table-v2] modify the type definition of the row slot (#14254)

* fix(docs): [table-v2] modify the type definition of the row slot

* Update table-v2.md

* Update table-v2.md

jiechen257 committed:
test(components): [button] fix typos in test description (#14314)

btea committed:
fix(docs): [form] `label-position` type error (#14312)

StephenKe committed:
fix(components): row-class-name bgColor can not cover fixed column (#14225)

makedopamine committed:
docs(form): add that `prop` supports dot notation string (#14325)

Mario34 committed:
fix(components): [tabs] Fix some events accidentally firing (#14224)

makedopamine committed:
refactor(descriptions): improve types (#14313)

* refactor(descriptions): improve types

* use buildProps

btea committed:
feat(components): [tabs] add addIcon slot (#12970)

* feat(components): [tabs] add addIcon slot

* docs: add example

* docs: update tabs.md

* chore: add version tag

guohenghai committed:
test(components): [teleport] Fixed spelling error ElTelport to ElTelepot (#14337)

In the teleport test file, changed "describe('ElTelport'" to "describe('ElTeleport'".

closed only spelling error

Co-authored-by: guohenghai <guohenghai@126.com>

Xaw-xu committed:
feat(components): [form-item]Label attribute for judging modification (#14344)

HeftyKoo committed:
refactor(components): [tabs] enhance instance type (#14352)

* refactor(components): [tabs] enhance instance type

1. exports Tabs instance type

2. add expose type to TabNavInstanec

* chore: remove useless import

mivui committed:
fix: cannot find type definition file for `element-plus/global` (#13698)

* fix:Cannot find type definition file for 'element-plus/global'

* Update package.json

* Update package.json

---------

Co-authored-by: btea <2356281422@qq.com>

btea committed:
refactor(components): [select] use utils function (#14345)

btea committed:
fix(components): [description] use `withDirectives` add custom directive (#14299)

* fix(components): [description] use `withDirectives` add custom directive

* feat: add DirectiveArguments

cuongle-hdwebsoft committed:
fix(components): fix: update timepicker options when changing locale (#14287)

* fix(components): fix: update timepicker options when changing locale

* fix(components): fix: update timepicker options when changing locale

* fix(components): fix: update timepicker options when changing locale

betavs committed:
fix(components): [popper] invalid when props z-index is zero (#14375)

zhousg committed:
chore(MessageBox): unused keyframe (#14335)

docs(MessageBox): unused keyframe

YunYouJun committed:
fix(theme-chalk): tabel tr bg in dark mode (#14443)

FrontEndDog committed:
fix(components): [date-picker] slot compatible with Vue3.3.x (#14354)

FrontEndDog committed:
chore(components): [table] fix typo (#14473)

chore(other): update explanatory note

FrontEndDog committed:
fix(components): [select-v2] read properties of null (#14321)

btea committed:
fix(play): apply esbuild plugin to tsx component (#14480)

wonderl17 committed:
fix(components): [color-picker] exposed show method wrong behavior (#14064)

fix(components): [color-picker] fix show method wrong behavior

TomatoDroid committed:
feat(components): watermark component (#14236)

* feat(components): watermark component

be able to set text,multi-text,image as watermark

* docs(components): update image watermark example

update the image for dark mode

betavs committed:
docs(components): [dialog] docs description abnormal (#14496)

btea committed:
docs(components): [watermark] update doc (#14500)

MrWeilian committed:
fix(components): [table] table-layout auto fixed by sticky (#11742)

* fix(components): [table] table-layout auto fixed by sticky

* Update packages/theme-chalk/src/table.scss

Co-authored-by: qiang <qw13131wang@gmail.com>

* fix(components): [table] add index variable for thead

* fix(components): [table] const z-index change to variable

* fix(components): [table] thead, scrolbar, border add z-index

* fix: remove duplicate property

---------

Co-authored-by: qiang <qw13131wang@gmail.com>

btea committed:
refactor(components): [checkbox] set the label default value to `undefined` (#14011)

* fix(components): [checkbox] allow the label value to be number 0

* fix: update computed

* test: add test case

* fix: update

MrWeilian committed:
fix(components): [carousel] two length transform optimize (#12174)

* fix(components): [carousel] two length transform optimize

* fix(components): [carousel] default init function slot

* fix(components): [carousel] cancel optimize when loop is false

* fix(components): [carousel] filter card type

* fix: test

* style: code lint

* fix: cardtype

* fix: remove test code

* fix: act index

* feat: use flattedchildren

* fix: code lint

* fix: rename and add comment

tolking committed:
feat(components): [color-picker] add focus and blur event (#14244)

* feat(components): [color-picker] add focus blur event

* docs: updata

wzc520pyfm committed:
docs(components): [descriptions] use new display tag (#12673)

kamesan012 committed:
fix(components): inline style doesn't use the custom sass setting value (#14117)

* fix(components): inline style doesn't use the custom sass setting value

closed #14038

* Update packages/components/select/src/useSelect.ts

Co-authored-by: btea <2356281422@qq.com>

* fix(components): update codes

* fix(components): use ns.cssVarName('input-height')

---------

Co-authored-by: btea <2356281422@qq.com>

FrontEndDog committed:
fix(components): [table] fix summary row cells on table with layout auto not aligned with data cells (#14315)

fix(components): [table] fix summary row cells not aligned

betavs committed:
fix(components): [upload] icon style abnormal (#14471)

btea committed:
feat(docs): theme switch use startViewTransition api (#14489)

* feat(docs): theme switch use startViewTransition api

* chore: format

* feat: update

tolking committed:
fix(components): [table] show-summary style error when table-layout=auto (#14523)

Karolis-Stoncius committed:
fix(components): [checkbox] indeterminate checkbox a11y fix (#14322)

tolking committed:
feat(components): [select] accessibility enhancement (#14503)

* feat: accessibility enhancement

* fix: adjusting the attributes of Scrollbar

* Update docs/en-US/component/scrollbar.md

Co-authored-by: btea <2356281422@qq.com>

* docs: updata

* fix(components): [select] aria-selected error

---------

Co-authored-by: btea <2356281422@qq.com>

IceyWu committed:
feat(components): [image-viewer] add `minScale` and `maxScale` (#14120)

* fix(components): [image-viewer] add scale props

* docs(components): [image] add min-scale and max-scale  properties

* feat(components): [image] add minScale and maxScale properties

* docs(components): [image] add example properties

* docs(components): [image] add docs  properties

* docs(components): [image] add docs properties

* docs(components): [image] update  version tag

btea committed:
docs: update version tag (#14532)

Lucky-Ya-Q committed:
fix(components): [sub-menu] style error in collapsed state (#13135)

* fix(components): [sub-menu] style error in collapsed state

* fix(theme-chalk): [menu] optimize

---------

Co-authored-by: tolking <qw13131wang@gmail.com>

tolking committed:
fix(components): [menu] remove excess inline styles (#13693)

* fix(components): [menu] remove excess inline styles

* fix(theme-chalk): [menu] fix styles

FrontEndDog committed:
feat(components): [date-picker] add dateFormat and timeFormat props (#14330)

* fix(components): [time-picker] extractDateFormat extractTimeFormat

* fix(components): [time-picker] Provide default return values

* feat(components): [date-picker] add dateFormat and timeFormat props

* feat(components): [date-picker] update demo

* Update docs/en-US/component/datetime-picker.md

Co-authored-by: btea <2356281422@qq.com>

* Update docs/en-US/component/datetime-picker.md

---------

Co-authored-by: qiang <qw13131wang@gmail.com>
Co-authored-by: btea <2356281422@qq.com>

gimjin committed:
fix(components): [text] not support multi-line ellipsis (#11976)

* docs(components): [text] line-clamp prop and update example

* test(components): [text] line-clamp test

* fix(components): [text] not support multi-line ellipsis

* docs: updata

---------

Co-authored-by: qiang <qw13131wang@gmail.com>

tolking committed:
refactor(components)!: [select-v2] value-key is used for object select (#13263)

refactor(components): [select-v2] value-key is used for object select

element-bot committed:
chore: Update changelog 2.4.0 (#14538)

chore: update changelog 2.4.0

0song committed:
chore: action memory (#14541)

btea committed:
docs(components): [watermark] update font (#14542)

sleepyShen1989 committed:
test(components): [alert] change test case to match description (#14551)

cuongle-hdwebsoft committed:
fix(components): inconsistent style rounded button (#14552)

fix(components): fix: inconsistent style rounded button

fix: inconsistent style rounded button

closed #14550

cuongle-hdwebsoft committed:
fix(components): prevent carousel from auto playing (#14553)

prevent carousel from auto playing

closed #14524

wzc520pyfm committed:
docs(components): [upload] use new display tag (#12701)

* docs(components): [upload] use new display tag

* docs(components): [upload] add description for beford-upload

* feat(components): [upload] delete useless attribute

* docs(components): [upload] adjust description and default for attr

* docs(components): [upload] add more detail for data attr

Karolis-Stoncius committed:
feat(components): [switch] add label attribute for accessibility (#14319)

* feat(components): [switch] add label attribute for accessibility

* refactor(components): [switch] set default switch label to undefined

* docs: version change

tolking committed:
fix(components): [watermark] missing the CSS file (#14564)

closed #14560

YunYouJun committed:
fix(theme-chalk): menu horizontal popover height, close #14566 (#14573)

tolking committed:
fix(components): [select] input height calculation error (#14565)

closed #14561

element-bot committed:
chore: Update changelog 2.4.1 (#14588)

chore: update changelog 2.4.1

baiwusanyu-c committed:
fix(components): [select] Fix border and shadow color error (#12757)

* fix(components): [select] Fix border and shadow color error

* fix(components): [select] updated code

* fix(components): [select] updated code

tolking committed:
feat(components): [select-v2] add props attribute (#14536)

* feat(components): [select-v2] add props attribute

* docs: updata

* docs: updata

* docs: updata

betavs committed:
fix(components): [message-box] uncaught on draggable (#14618)

aafnnp committed:
feat(i18n): add Swahili(sw) translation (#14660)

Co-authored-by: pfan <fan.pan@palmpay-inc.com>

tolking committed:
fix(components): [select-v2] label error when value not in the options (#14656)

closed #14621

cc-hearts committed:
fix(components): [rate] Fix the background rate when decimal (#14622)

* fix(components): rate

fix the background rate when decimal

closed #14619

* test: add test case

IceyWu committed:
feat(components): [watermark] add font textAlign and textBaseline (#14567)

nothing-sy committed:
fix(components): [dialog] fix el-dialog z-index bug (#14373)

* Fix el-dialog z-index bug

* Update packages/components/dialog/src/use-dialog.ts

Co-authored-by: btea <2356281422@qq.com>

* chore: lint error

---------

Co-authored-by: qiang <qw13131wang@gmail.com>
Co-authored-by: btea <2356281422@qq.com>

Jungzl committed:
fix(components): [dialog] consistent imports of lodash

yujinpan committed:
fix(components): [tree-select] v-model invalid when source changes (#14603)

sleepyShen1989 committed:
docs(components): [autocomplete] fix website display (#14557)

* docs(components): [autocomplete] fix website display

* docs(components): [autocomplete]add column of Type in Slots

xingyixiang committed:
fix(components): [table] tooltipOptions.showAfter is not effective (#13175)

* fix(components): [table]
tooltipOptions.showAfter is not effective

* fix(components): [table]
tooltipOptions.showAfter is not effective

---------

Co-authored-by: btea <2356281422@qq.com>

btea committed:
Revert "fix(components): [table] tooltipOptions.showAfter is not effe… (#14735)

Revert "fix(components): [table] tooltipOptions.showAfter is not effective (#13175)"

This reverts commit 6238ec8456095b369fcf6e81a79440314cf774d5.

element-bot committed:
chore: Update changelog 2.4.2 (#14737)

chore: update changelog 2.4.2

faga295 committed:
feat(components): [card] adds footer slot (#10357)

feat(components): [card]

adds footer slot

closed #6873

Co-authored-by: btea <2356281422@qq.com>

dannyhebertbell committed:
chore: update sync fork

dannyhebertbell committed:
chore: revert pnpm-lock change

tolking commented:
Thanks for your contribution.

This PR contains a lot of other git commit information, which makes the review very difficult. If possible, I suggest you create a new PR.
